### PR TITLE
addpkg(x11/qbittorrent{,-nox}): 4.6.4

### LINKED ISSUE #10081: [Package]: Qbittorrent-Nox
### Package description

a command line Bittorrent client written in C++ / Qt

### Home page URL

https://www.qbittorrent.org/

### Source code URL

https://github.com/userdocs/qbittorrent-nox-static

### Packaging policy acknowledgement

- [X] I have read and understand the [Packaging Policy](https://github.com/termux/termux-packages/blob/master/CONTRIBUTING.md#packaging-policy).

### Additional information

_No response_

### PR DESCRIPTION
**Pull request by TomJo2000**
closes:
- #10081

`qbittorrent` (even the headless `-nox` version) depends on `qt5-qtbase`, so it needs to be in the X11 Repo.

I'm forcing Cmake for the build.
Cmake has better [upstream build instructions](https://github.com/qbittorrent/qBittorrent/wiki/Compilation-with-CMake:-common-information).
Builds fine with `make` as well, if that's preferable.

---

If there's an easy way to just build it twice I'd actually like to build it once with GUI and once without,
and split off `qbittorrent-nox` into a subpackage.
But I couldn't figure out a good way to do that.
And the request was for the headless one.

### CODE REVIEW

File: x11-packages/qbittorrent-nox/build.sh
Comment (licy183) [inline x11-packages/qbittorrent-nox/build.sh:18]: This is set automatically by the building system.

https://github.com/termux/termux-packages/blob/8691c1f3a1adaf31ed1e6dc46c3e9fb1f2263ead/scripts/build/configure/termux_step_configure_cmake.sh#L58

Comment (TomJo2000) [reply x11-packages/qbittorrent-nox/build.sh:18]: Not sure how I didn't catch that.

Comment (TomJo2000) [reply x11-packages/qbittorrent-nox/build.sh]: > How do you like to squash commits? Let me know the method

Usually `git commit --amend`, followed by `git push --force-with-lease=branchname`
which works great if you just wanna modify the latest commit.

---

If a PR ends up with multiple commits I usually do an "interactive rebase" with `git rebase -i HEAD~42`
(where `42` gets swapped out for the number of commits I actually need to work on)
I can then reorder, squash or "fixup" ***my own*** commits.
But crucially you do not mess with commits already on the master branch.

---

"Fixup" commits are another option if you are a bit more organized than me.
In that case you commit for "fixup" commits using `git commit --fixup=ref`,
where `ref` is the commit you want to squash them into later.
- Example: 
`e08e34581` is the commit you want to fixup, so you can commit your new changes with,
`git commit --fixup=e08e34581`
Which will then generate a commit along the lines of:
```styl
[qbittorrent 585e9e714] fixup! addpkg(x11/qbittorrent-nox): 4.6.4 closes #10081
```
These can be automatically squashed before pushing using `git rebase --autosquash`, this can be combined with an interactive rebase if you like.

Comment (truboxl) [reply x11-packages/qbittorrent-nox/build.sh]: I still see -DCMAKE_INSTALL_PREFIX not removed...

**Comment by truboxl** [comment]
Auto update is not working
```console
INFO: Updating qbittorrent-nox [Current version: 4.6.4]
ERROR: No 'latest-release-tag' found (https://api.github.com/repos/qbittorrent/qBittorrent/releases/latest).
Try using 'newest-tag'.
```

**Review by truboxl — APPROVED** [review]
> We probably don't want it to even try creating an issue from a failing dry-run, even though it can't

Ah. It should be fine since I purposely disabled permission to write issue. I leave it to others to fix it if they find it annoying.

**Comment by TomJo2000** [comment]
By the way, I mentioned it above but I think it might have gotten lost.

### Is there any easy way I could hook the build process to build `qbittorrent` twice - once with and once without GUI - so we can split off `qbittorrent-nox` into a subpackage?

**Comment by truboxl** [comment]
https://github.com/termux/termux-packages/blob/6c51913c4f47aff7dd5263d6442b65cd716874a2/packages/libncnn/build.sh#L70

Can try something like this and then add a new subpackage with its own dependencies without depend on parent.
If it doesn't work can always split into 2 packages like vlc and vlc-qt.

**Comment by TomJo2000** [comment]
"Ol' reliable" is of course an option, but a subpackage feels more appropriate.

**Comment by TomJo2000** [comment]
Alright, that oughta do it.

**Comment by TomJo2000** [comment]
- [x] DEBs look as expected
- [x] autoupdate function is passing
- [x] subpackage is being built and packaged correctly
- [x] Works on device (Tested with a Fairphone 5, Android 13, AArch64)
- [x] Commits are squashed

I think we're good to merge, I'll wait for any last minute comments, otherwise I'll merge this in 8 to 10 hours after I get some sleep.

**Pull request merged**
